UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/AzureBlobStorageClient.cs

```diff
@@ -0,0 +1,110 @@
+        private readonly ILogger<AzureBlobStorageClient> logger;
+        public AzureBlobStorageClient(IScsStorageService scsStorageService, IOptions<EssFulfilmentStorageConfiguration> storageConfig, IAzureMessageQueueHelper azureMessageQueueHelper, ILogger<AzureBlobStorageClient> logger)
+        {
+            this.scsStorageService = scsStorageService;
+            this.storageConfig = storageConfig;
+            this.azureMessageQueueHelper = azureMessageQueueHelper;
+            this.logger = logger;
+        }
+
+        public async Task<bool> StoreScsResponseAsync(string containerName, string batchId, SalesCatalogueResponse salesCatalogueResponse, CancellationToken cancellationToken)
+        {
+            string uploadFileName = string.Concat(batchId, ".json");
+
+            string storageAccountConnectionString =
+                  scsStorageService.GetStorageAccountConnectionString();
+            CloudBlockBlob cloudBlockBlob = GetCloudBlockBlob(uploadFileName, storageAccountConnectionString, containerName);
+            cloudBlockBlob.Properties.ContentType = CONTENT_TYPE;
+
+            await UploadScsResponseToBlobAsync(cloudBlockBlob, salesCatalogueResponse);
+
+            await AddQueueMessage(batchId, salesCatalogueResponse, cloudBlockBlob);
+
+            logger.LogInformation(EventIds.SCSResponseStoredAndSentMessageInQueue.ToEventId(), "Sales catalogue response saved for the {batchId}", batchId);
+            return true;
+        }
+
+        public async Task AddQueueMessage(string batchId, SalesCatalogueResponse salesCatalogueResponse, CloudBlockBlob cloudBlockBlob)
+        {
+            ScsResponseQueueMessage scsResponseQueueMessage = GetScsResponseQueueMessage(batchId, salesCatalogueResponse, cloudBlockBlob);
+            var scsResponseQueueMessageJSON = JsonConvert.SerializeObject(scsResponseQueueMessage);
+            await azureMessageQueueHelper.AddMessage(storageConfig.Value, scsResponseQueueMessageJSON);
+        }
+
+        public async Task UploadScsResponseToBlobAsync(CloudBlockBlob cloudBlockBlob , SalesCatalogueResponse salesCatalogueResponse)
+        {
+            var serializeJsonObject = JsonConvert.SerializeObject(salesCatalogueResponse);            
+
+            using (var ms = new MemoryStream())
+            {
+                LoadStreamWithJson(ms, serializeJsonObject);
+                await cloudBlockBlob.UploadFromStreamAsync(ms);
+            }
+        }
+
+        private ScsResponseQueueMessage GetScsResponseQueueMessage(string batchId, SalesCatalogueResponse salesCatalogueResponse, CloudBlockBlob cloudBlockBlob)
+        {
+            int fileSize = GetFilesize(salesCatalogueResponse);
+            var scsResponseQueueMessage = new ScsResponseQueueMessage()
+            {
+                BatchId = batchId,
+                ScsResponseUri = cloudBlockBlob.Uri.AbsoluteUri,
+                FileSize = fileSize
+            };
+            return scsResponseQueueMessage;
+        }
+
```

Comment:
Could be GetSalesCatalogueServiceResponseQueueMessage

Reply:
Done